src/adapter.rs

```diff
@@ -471,6 +473,118 @@ impl Adapter {
+                    winapi::shared::ws2def::AF_INET => {
+                        let octets = endpoint.Ipv4.sin_addr.S_un.S_un_b;
+                        let address = Ipv4Addr::new(octets.s_b1, octets.s_b2, octets.s_b3, octets.s_b4);
+                        let port = u16::from_be(endpoint.Ipv4.sin_port);
+                        SocketAddr::V4(SocketAddrV4::new(address, port))
+                    }
+                    winapi::shared::ws2def::AF_INET6 => {
+                        let octets = endpoint.Ipv6.sin6_addr.u.Byte;
+                        let address = Ipv6Addr::from(octets);
+                        let port = u16::from_be(endpoint.Ipv6.sin6_port);
+                        let flow_info = endpoint.Ipv6.sin6_flowinfo;
+                        let scope_id = endpoint.Ipv6.__bindgen_anon_1.sin6_scope_id;
+                        SocketAddr::V6(SocketAddrV6::new(address, port, flow_info, scope_id))
+                    }
+                    _ => { panic!("Illegal address family {}", endpoint.si_family); }
+                };
+                let mut wg_peer = WireguardPeer {
+                    Flags: peer.Flags as u32,
+                    PublicKey: peer.PublicKey,
+                    PresharedKey: peer.PresharedKey,
+                    PersistentKeepalive: peer.PersistentKeepalive,
+                    Endpoint: endpoint,
+                    TxBytes: peer.TxBytes,
+                    RxBytes: peer.RxBytes,
+                    LastHandshake: peer.LastHandshake,
+                    AllowedIps: Vec::with_capacity(peer.AllowedIPsCount as usize),
+                };
+                for _ in 1..=peer.AllowedIPsCount {
+                    let allowed_ip: WIREGUARD_ALLOWED_IP = reader.read();
+                    let prefix_length = allowed_ip.Cidr;
+                    let allowed_ip = match allowed_ip.AddressFamily as i32 {
+                        winapi::shared::ws2def::AF_INET => {
+                            let octets = allowed_ip.Address.V4.S_un.S_un_b;
+                            let address = Ipv4Addr::new(octets.s_b1, octets.s_b2, octets.s_b3, octets.s_b4);
+                            IpNet::V4(Ipv4Net::new(address, prefix_length).expect("prefix is valid"))
+                        }
+                        winapi::shared::ws2def::AF_INET6 => {
+                            let octets = allowed_ip.Address.V6.u.Byte;
+                            let address = Ipv6Addr::from(octets);
+                            IpNet::V6(Ipv6Net::new(address, prefix_length).expect("prefix is valid"))
+                        }
+                        _ => { panic!("Illegal address family {}", allowed_ip.AddressFamily); }
+                    };
+                    wg_peer.AllowedIps.push(allowed_ip);
+                }
+                wg_interface.Peers.push(wg_peer);
+            }
+            wg_interface
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct WireguardPeer {
+    #[doc = "< Bitwise combination of flags"]
```

Comment:
TBH I just copy-pasted the auto-generated struct from `wireguard_nt_raw.rs`. I'll tidy it up a bit. Once this is merged in, maybe we can think of how to merge some of these structs together (e.g. have just one struct instead of `SetPeer` and `WireguardPeer`)